File: Programming_in_Python/Leetcode_Template/list_node.py
from typing import List



class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


def list_deserializer(nums: List[int]) -> ListNode:
    head = current = ListNode()
    for i in nums:
        current.next = ListNode(i)
        current = current.next
    return head.next

def list_serializer(root: ListNode) -> List[int]:
    current = root
    ans = []
    while current is not None:
        ans.append(current.val)
        current = current.next
    return ans

File: Programming_in_Python/Solutions/Leetcode_Java/LeetcodeDefaultImplementationsOfDS/ListNode.java
package LeetcodeDefaultImplementationsOfDS;

public class ListNode {
    public int val;
    public ListNode next;
    public ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
    public void printer(ListNode head) {
        while(head != null) {
            System.out.print(head.val + "->");
            head = head.next;
        }
        System.out.println("null");
        System.out.println();
    }
}